beacon/pow/src/main/java/tech/pegasys/teku/beacon/pow/Eth1DepositManager.java

```diff
@@ -118,31 +117,29 @@ public void start() {
   }
 
   private SafeFuture<LoadDepositSnapshotResult> loadDepositSnapshot() {
-    final LoadDepositSnapshotResult fileDepositSnapshotResult =
-        depositSnapshotFileLoader.loadDepositSnapshot();
-    if (fileDepositSnapshotResult.getDepositTreeSnapshot().isPresent()
-        && !takeBestDepositSnapshot) {
-      LOG.debug("Deposit snapshot from file is provided, using it");
-      return SafeFuture.completedFuture(fileDepositSnapshotResult);
+    if (customDepositSnapshotPathPresent) {
+      LOG.debug("Custom deposit snapshot is provided, using it");
+      return SafeFuture.completedFuture(depositSnapshotFileLoader.loadDepositSnapshot());
     }
 
-    if (takeBestDepositSnapshot) {
-      STATUS_LOG.loadingDepositSnapshotFromDb();
-      return depositSnapshotStorageLoader
-          .loadDepositSnapshot()
-          .thenApply(
-              storageDepositSnapshotResult -> {
-                if (storageDepositSnapshotResult.getDepositTreeSnapshot().isPresent()) {
-                  final DepositTreeSnapshot storageSnapshot =
-                      storageDepositSnapshotResult.getDepositTreeSnapshot().get();
-                  STATUS_LOG.onDepositSnapshot(
-                      storageSnapshot.getDepositCount(), storageSnapshot.getExecutionBlockHash());
-                }
-                return ObjectUtils.max(storageDepositSnapshotResult, fileDepositSnapshotResult);
-              });
-    } else {
-      return depositSnapshotStorageLoader.loadDepositSnapshot();
-    }
+    STATUS_LOG.loadingDepositSnapshotFromDb();
+    return depositSnapshotStorageLoader
+        .loadDepositSnapshot()
+        .thenApply(
+            storageDepositSnapshotResult -> {
+              if (storageDepositSnapshotResult.getDepositTreeSnapshot().isPresent()) {
+                final DepositTreeSnapshot storageSnapshot =
+                    storageDepositSnapshotResult.getDepositTreeSnapshot().get();
+                STATUS_LOG.onDepositSnapshot(
+                    storageSnapshot.getDepositCount(), storageSnapshot.getExecutionBlockHash());
+                return storageDepositSnapshotResult;
+              }
+
+              final LoadDepositSnapshotResult fileDepositSnapshotResult =
```

Comment:
It would be nice to have a STATUS_LOG.loadingDepositSnapshotFromFile(filename) when we load from file.

We could even log something like: "No deposit snapshot found on local db, loading it from file x" or something? What do you think?

Reply:
I'm not sure if we need to say no snapshot found on local db? But can add that if you aren't satisfied with what I have done